### DISCUSSION TITLE
How to Convert .AFP (Advanced Function Presentation) files to PDF using java or python?

### DISCUSSION THREAD

**Discussion by AtharvaBallal** [community/community#147955] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello,

I want to convert multiple AFP files to PDF in one go and I want to do this in Python can anyone know the solution?  
Please help me.

**Comment by hostbett** [comment]
To convert AFP (.AFP) files to PDF, you can use libraries like PDFTron in Java, which offers a simple API to load and save AFP as PDF. In Python, tools like Ghostscript can be invoked using the subprocess module to convert AFP files to PDF by running a command. Alternatively, the AFP2PDF library, if available, can also handle AFP to PDF conversion directly.

**Reply by AtharvaBallal** [reply]
@hostbett Thank you for the response can you please provide more updates on the Ghostscript like where I can find that and also same for AFP2PDF it will be a great help for me.

**Comment by SinnConsultants** [comment]
TallPine Technologies provides a command line and API tool to help. The largest challenge you may have is ensuring your AFP is in a stardard format that can be converted. If you have any fidelity issues, reach out with questions to https://tallpine.com
